sqlmesh/core/engine_adapter/athena.py

```diff
@@ -0,0 +1,441 @@
+
+        is_hive = self._table_type(table_properties) == "hive"
+        is_iceberg = not is_hive
+
+        if is_hive and not expression:
+            # Hive tables are CREATE EXTERNAL TABLE, Iceberg tables are CREATE TABLE
+            # Unless it's a CTAS, those are always CREATE TABLE
+            properties.append(exp.ExternalProperty())
+
+        if table_description:
+            properties.append(exp.SchemaCommentProperty(this=exp.Literal.string(table_description)))
+
+        if partitioned_by:
+            schema_expressions: t.List[exp.Expression] = []
+            if is_hive and columns_to_types:
+                # For Hive-style tables, you cannot include the partitioned by columns in the main set of columns
+                # In the PARTITIONED BY expression, you also cant just include the column names, you need to include the data type as well
+                # ref: https://docs.aws.amazon.com/athena/latest/ug/partitions.html
+                for match_name, match_dtype in self._find_matching_columns(
+                    partitioned_by, columns_to_types
+                ):
+                    column_def = exp.ColumnDef(this=exp.to_identifier(match_name), kind=match_dtype)
+                    schema_expressions.append(column_def)
+            else:
+                schema_expressions = partitioned_by
+
+            properties.append(
+                exp.PartitionedByProperty(this=exp.Schema(expressions=schema_expressions))
+            )
+
+        if clustered_by:
+            # Athena itself supports CLUSTERED BY, via the syntax CLUSTERED BY (col) INTO <n> BUCKETS
+            # However, SQLMesh is more closely aligned with BigQuery's notion of clustering and
+            # defines `clustered_by` as a List[str] with no way of indicating the number of buckets
+            #
+            # Athena's concept of CLUSTER BY is more like Iceberg's `bucket(<num_buckets>, col)` partition transform
+            logging.warning("clustered_by is not supported in the Athena adapter at this time")
+
+        if storage_format:
+            if is_iceberg:
+                # TBLPROPERTIES('format'='parquet')
+                table_properties["format"] = exp.Literal.string(storage_format)
+            else:
+                # STORED AS PARQUET
+                properties.append(exp.FileFormatProperty(this=storage_format))
+
+        if table and (location := self._table_location(table_properties, table)):
+            properties.append(location)
+
+        for name, value in table_properties.items():
+            properties.append(exp.Property(this=exp.var(name), value=value))
+
+        if is_iceberg and expression:
+            # To make a CTAS expression persist as iceberg, alongside setting `table_type=iceberg` (which the user has already
+            # supplied in physical_properties and is thus set above), you also need to set:
+            #  - is_external=false
+            #  - table_location='s3://<path>'
```

Comment:
Should we ensure somehow that the location has already been set at this point?

Reply:
Oh nice catch!

Reply:
Actually, the location will be set already a few lines up if the user supplied it (or `s3_warehouse_location` was set in the config).

But ive just done some tests and unlike Trino, it looks like Athena will not automatically generate table locations for you if the schema the table is in was created with a location set.

I created a schema using `CREATE SCHEMA foo LOCATION 's3://path'` and then tried to create both Hive and Iceberg tables in that schema without setting a location explicitly. Both times it failed with an error asking to set the location.

So i'll tighten this up and throw an error if SQLMesh cant figure out the table location